bots/DiscordBot/PublicCommands.js

```diff
@@ -0,0 +1,357 @@
+                    }
+                });
+            });
+        });
+    }
+
+    async help(msg, directions) {
+        //!help <command>
+        let id = msg.author.id;
+        let response = `Hey <@${id}>,\n`;
+        if (directions.length == 0) {
+            response += this.botResponseService.helpMain;
+            msg.channel.send(response);
+        } else {
+            switch (directions[0]) {
+                case 'gameinfo':
+                    response += this.botResponseService.helpGameinfo;
+                    msg.channel.send(response);
+                    break;
+                case 'help':
+                    response += this.botResponseService.helpHelp;
+                    msg.channel.send(response);
+                    break;
+                case 'leaderboard_global':
+                    response += this.botResponseService.helpLeaderboard_global
+                    msg.channel.send(response);
+                    break;
+                case 'leaderboard_local':
+                    response += this.botResponseService.helpLeaderboard_local
+                    msg.channel.send(response);
+                    break;
+                case 'userinfo':
+                    response += this.botResponseService.helpUserinfo
+                    msg.channel.send(response);
+                    break;
+                default:
+                    response += this.botResponseService.helpUnidentified
+                    msg.channel.send(response);
+            }
+        }
+    }
+
+    async leaderboard_global(msg, directions) {
+        //!leaderboard_global <filter> (<page>)
+
+        // Calculating how the leaderboard looks
+        let sortingKey = directions[0];
+        let limit = await this.userService.getUserCount();
+        let result = await this.leaderboardService.getLeaderboard(limit, sortingKey);
+        let leaderboard = result.leaderboard;
+        let pageCount = Math.ceil(leaderboard.length / 20)
+
+        // here be dragons
+        const getNestedObject = (nestedObj, pathArr) => {
+            return pathArr.reduce((obj, key) =>
+                (obj && obj[key] !== 'undefined') ? obj[key] : -1, nestedObj)
+        }
+
+        const generateLeaderboard = page => {
+            let position_list = "";
+            let username_list = "";
+            let sortingKey_list = "";
+            let lowerLimit = (page - 1) * 20 + 1
+            let upperLimit = page * 20
+            for (let i = lowerLimit; i < upperLimit; i++) {
+                if (!leaderboard[i]) { break; }
+                position_list += leaderboard[i].position + "\n";
+                username_list += leaderboard[i].username + "\n";
+                sortingKey_list += getNestedObject(leaderboard[i], result.sorter.fullKey.split('.')) + "\n"
+            }
+            let response = await this.botResponseService.leaderboard_global(page, sortingKey, position_list, username_list, sortingKey_list)
```

Comment:
As we are now paging the bot response, we should select only the number of records a page can display from the database. i.e If a page displays 10, then we should retrieve only 10 records.

See MongoDB `skip` and `limit`